app/src/main/java/com/github/nkzawa/socketio/androidchat/MainFragment.java

```diff
@@ -259,6 +266,37 @@ private void scrollToBottom() {
         mMessagesView.scrollToPosition(mAdapter.getItemCount() - 1);
     }
 
+    private Emitter.Listener onConnect = new Emitter.Listener() {
+        @Override
+        public void call(Object... args) {
+            getActivity().runOnUiThread(new Runnable() {
+                @Override
+                public void run() {
+                    if(!isConnected) {
+                        mSocket.emit("add user", mUsername);
```

Comment:
I think we have to check if `mUsername` exists ? since it would be `null` before login.